### PR TITLE
Add API to ping OIDC endpoint

### PR DESCRIPTION
**Pull request by reasonerjt**
This commit adds an API to help admin verify the OIDC endpoint is a
valid one.

Signed-off-by: Daniel Jiang <jiangd@vmware.com>

### CODE REVIEW

File: docs/swagger.yaml [hunk 1]
```diff
@@ -3478,6 +3478,39 @@ paths:
...
+            properties:
+              url:
+                type: string
+                description: The URL of OIDC endpoint to be tested.
+              verify_cert:
+                type: boolean
+                description: Whether the certificate should be verified
+      responses:
+        '200':
+          description: The specified robot account is successfully deleted.
```
Comment (wy65701436) [inline docs/swagger.yaml]: please fix the description.

File: docs/swagger.yaml [hunk 2]
```diff
@@ -3478,6 +3478,39 @@ paths:
...
+        '400':
+          description: The ping failed
+        '401':
+          description: User need to log in first.
+        '403':
+          description: User does not have permission to call this API
+        '500':
```
Comment (wy65701436) [inline docs/swagger.yaml]: and there is no 500 return in the API.

File: docs/swagger.yaml (hunk 1 above)
Comment (stonezdj) [inline docs/swagger.yaml]: It should be Ping OIDC success

File: src/core/api/oidc.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+func (oa *OIDCAPI) Ping() {
+	var c oidc.Conn
+	if err := oa.DecodeJSONReq(&c); err != nil {
+		log.Error("Failed to decode JSON request.")
+		oa.SendBadRequestError(err)
+		return
+	}
+	if err := oidc.TestEndpoint(c); err != nil {
+		log.Errorf("Failed to verify connection: %+v, err: %v", c, err)
+		oa.SendBadRequestError(err)
```
Comment (wy65701436) [inline src/core/api/oidc.go]: what's the difference between there two BadRequest? Does the second one mean the endpoint is not reachable?

Comment (reasonerjt) [reply src/core/api/oidc.go]: yes, kind of, either unreachable, or it's not an OIDC provider.

File: docs/swagger.yaml (hunk 1 above)
Comment (reasonerjt) [reply docs/swagger.yaml]: Will do

File: docs/swagger.yaml (hunk 2 above)
Comment (reasonerjt) [reply docs/swagger.yaml]: Oh, yes, will do.

**Pull request merged**